### PR TITLE
Feature/114 gh action test runner

### LINKED ISSUE #114: Github Actions Test Runner
### Context

We should run our API tests (for starters) on GH Actions for regression testing.

### Actions

- [x] Add the test run after the existing actions in `.github/workflows/ecr_api.yml`
- [x] Create Docker Compose file with Postgres & Existing API Docker container
- [x] Start Postgres/Postgis container & configure the API to use it
- [x] Run `npm test` & `npm run lint` on the docker image

Similiar Repo with GH Actions: https://github.com/developmentseed/lulc-infra/blob/develop/.github/workflows/api_test.yml

cc/ @srmsoumya

### PR DESCRIPTION
**Pull request by srmsoumya**
Fix #114 

Use Github Actions for API testing & linting.

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -8,19 +8,22 @@ COPY ./ $HOME/api
...
-    && DEBIAN_FRONTEND=noninteractive apt-get install -y nginx curl
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y git nginx curl
 
 RUN export NODEV='16.9.0' \
     && curl "https://nodejs.org/dist/v${NODEV}/node-v${NODEV}-linux-x64.tar.gz" | tar -xzv \
     && cp ./node-v${NODEV}-linux-x64/bin/node /usr/bin/ \
     && ./node-v${NODEV}-linux-x64/bin/npm install -g npm@7
 
 RUN npm install \
+    && npm uninstall bcrypt && npm install bcrypt \
```
Comment (batpad) [inline api/Dockerfile:19]: Could you add a comment or so saying why you needed to do this? (Even if its something like "I dont know why, but I needed to do this to make it work.") - will make it helpful in the future if someone wants to remove it / is wondering why it's there.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      - PORT=2001
+      - POSTGRES=postgres://docker:docker@postgis:5432/gis
+      - MAPBOX_TOKEN
+    volumes:
+      - /home/mle/api/node_modules
+
+  api:
+    <<: *base
+    container_name: ml-enabler-api
+    command: sh -c "sleep 5; npm run dev"
```
Comment (batpad) [inline docker-compose.yml:25]: do we know why we have these `sleep`s in here? Just curious.

File: api/Dockerfile (hunk 1 above)
Comment (srmsoumya) [reply api/Dockerfile:19]: > Error: /app/node_modules/bcrypt/lib/binding/bcrypt_lib.node: invalid ELF header

I was getting this error in my local docker setup, after reading I figured out it has to do something with how `bcrypt` compiles across different OS.  Here are the links that I referred to make it work [link1](https://www.richardkotze.com/top-tips/install-bcrypt-docker-image-exclude-host-node-modules) & [link2](https://medium.com/hacktive-devs/the-bcrypt-bg-on-docker-9bc36cc7f684).

File: docker-compose.yml (hunk 2 above)
Comment (srmsoumya) [reply docker-compose.yml:25]: As all these containers are created in parallel, this will allow time for PostGIS DB to be created & populated.

**Comment by ingalls** [comment]
@srmsoumya For getting this put together, I modified the code to work with the latest version of MLEnabler and have it deployed and running

**Pull request closed without merging**